adafruit_json_stream.py

```diff
@@ -39,16 +40,23 @@ def read(self):
         self.i += 1
         return char
 
-    def fast_forward(self, closer):
+    def fast_forward(self, closer, buffer=None):
         """Read through the stream until the character is ``closer``, ``]``
         (ending a list) or ``}`` (ending an object.) Intermediate lists and
         objects are skipped."""
+
         closer = ord(closer)
         close_stack = [closer]
         count = 0
         while close_stack:
             char = self.read()
             count += 1
+            if buffer:
+                if count == len(buffer):
+                    new_buffer = bytearray(len(buffer) + 32)
+                    new_buffer[: len(buffer)] = buffer
+                    buffer = new_buffer
```

Comment:
Why pass in buffer if you are willing to allocate a new one internally? Do you really just want a bool to say if you want the object back?

Reply:
The initial character needs to be set. It's already been read so it knows if it's a list or dict. Open to suggestions

Reply:
Ah, I see. I think it'd be better to change the argument to "return_object". That'd make the `if` tests much clearer. We can infer the start character from `closer` right? So we can make the buffer internally.

I also wonder if it is worth avoiding holding the whole object as a continuous string. I think there is a way to have `json` load from a file like object instead and read character by character.

Reply:
So `return_object` makes sense, and yes can infer.

So we can't do a `json` load, because we are reading forward and both don't have the first char and we aren't reading to the end - we need to find that part.

Open to ideas if I'm missing something

Reply:
And are you good with this:
```py
        buffer = None
        if return_object:
            buffer = bytearray(32)
            # ] = 93, [ = 91
            # } = 125, { = 123
            buffer[0] = closer - 2
```

Reply:
Yup, fine with me.

Reply:
@tannewt updated here: fd5d8a45293c148b6fd0a9ef6d29851a5ea06ecc